koschei/resolver.py

```diff
@@ -307,20 +204,28 @@ def update_dependency_changes(self, changes):
     def run(self, repo_id):
         start = time.time()
         self.log.info("Generating new repo")
+        self.repo_cache.prefetch_repo(repo_id)
         packages = self.get_packages(require_build=True)
         repo = Repo(repo_id=repo_id)
-        self.prepare_sack(repo_id)
-        if not self.sack:
+        # TODO repo_id
+        self.group = util.get_build_group(self.koji_session)
+        brs = util.get_rpm_requires(self.koji_session,
+                                    [dict(name=p.name,
+                                          version=p.last_complete_build.version,
+                                          release=p.last_complete_build.release,
+                                          arch='src') for p in packages])
+        sack = self.repo_cache.get_sack(repo_id)
```

Comment:
Have you considered using contextmanager to get resource scoping (not have to call `release_sack` before every return)? It would then look like this:
````python
with self.repo_cache.get_sack(repo_is) as sack:
     if blah:
        do stuff
        return
     do stuff
````